native/src/main/java/io/ballerina/protoc/builder/syntaxtree/utils/ClientUtils.java

```diff
@@ -0,0 +1,516 @@
+                        SyntaxTreeConstants.SYNTAX_TREE_GRPC_ERROR
+                )
+        );
+        VariableDeclaration sClient = new VariableDeclaration(
+                getTypedBindingPatternNode(
+                        getQualifiedNameReferenceNode("grpc", "StreamingClient"),
+                        getCaptureBindingPatternNode("sClient")
+                ),
+                getCheckExpressionNode(
+                        getRemoteMethodCallActionNode(
+                                getFieldAccessExpressionNode("self", "grpcClient"),
+                                methodName,
+                                new String[]{"\"" + method.getMethodId() + "\""}
+                        )
+                )
+        );
+        function.addVariableStatement(sClient.getVariableDeclarationNode());
+        function.addReturnStatement(getExplicitNewExpressionNode(clientName, new String[]{"sClient"}));
+        function.addQualifiers(new String[]{"isolated", "remote"});
+        return function;
+    }
+
+    public static Class getStreamingClientClass(Method method) {
+        String name = method.getMethodName().substring(0, 1).toUpperCase() + method.getMethodName().substring(1) +
+                "StreamingClient";
+        Class streamingClient = new Class(name, true);
+        streamingClient.addQualifiers(new String[]{"client"});
+
+        streamingClient.addMember(
+                getObjectFieldNode(
+                        "private",
+                        new String[]{},
+                        getQualifiedNameReferenceNode("grpc", "StreamingClient"),
+                        "sClient"));
+
+        streamingClient.addMember(getInitFunction().getFunctionDefinitionNode());
+
+        streamingClient.addMember(getSendFunction(method).getFunctionDefinitionNode());
+
+        streamingClient.addMember(getSendContextFunction(method).getFunctionDefinitionNode());
+
+        streamingClient.addMember(getReceiveFunction(method).getFunctionDefinitionNode());
+
+        streamingClient.addMember(getReceiveContextFunction(method).getFunctionDefinitionNode());
+
+        streamingClient.addMember(getSendErrorFunction().getFunctionDefinitionNode());
+
+        streamingClient.addMember(getCompleteFunction().getFunctionDefinitionNode());
+
+        return streamingClient;
+    }
+
+    private static Function getInitFunction() {
+        Function function = new Function("init");
+        function.addRequiredParameter(
+                TypeDescriptor.getQualifiedNameReferenceNode("grpc", "StreamingClient"),
+                "sClient"
+        );
+        function.addAssignmentStatement(
+                getFieldAccessExpressionNode("self", "sClient"),
+                getSimpleNameReferenceNode("sClient")
+        );
+        function.addQualifiers(new String[]{"isolated"});
+        return function;
+    }
+
+    private static Function getSendFunction(Method method) {
+        String inputCap;
+        switch (method.getInputType()) {
```

Comment:
Can't we move to another method to remove [duplicates](https://github.com/ballerina-platform/protoc-tools/pull/1/files#diff-f335938c6a0051a6339c6d27b866c0af2dfb229e530dded64093cd5bdef8416bR189)?